### ISSUE TITLE
Inline macro attribute parsing changes after first document conversion

### ISSUE THREAD

**Issue by wilkinsona** [asciidoctor/asciidoctorj#926]
There appears to be a problem with the parsing of an inline macro's attributes. Consider an inline macro that's used like this:

```
example:alpha.bravo[format=test]
```

I expect the `InlineMacroProcessor` implementation to be called with a map containing `format` -> `test`. This is the case the first time `convert` or `convertFile` is called. When a subsequent conversion is performed the behaviour of the parsing changes and the map then contains `text` -> `format=test`.

The problem is illustrated by this test:

```java
package com.example;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

import org.asciidoctor.Asciidoctor;
import org.asciidoctor.ast.ContentNode;
import org.asciidoctor.extension.InlineMacroProcessor;
import org.junit.jupiter.api.Test;

class AttributeTests {
	
	@Test
	void inlineMacroAttributes() {
		Asciidoctor asciidoctor = Asciidoctor.Factory.create();
		asciidoctor.javaExtensionRegistry().inlineMacro(new InlineMacroProcessor("example") {
			
			@Override
			public Object process(ContentNode parent, String target, Map<String, Object> attributes) {
				return createPhraseNode(parent, "quoted", attributes.toString(), attributes, new HashMap<>());
			}
			
		});
		assertThat(convert(asciidoctor)).contains("{format=test}");
		assertThat(convert(asciidoctor)).contains("{format=test}");
	}
	
	private String convert(Asciidoctor asciidoctor) {
		return asciidoctor.convert("example:alpha.bravo[format=test]", Collections.emptyMap());
	}

}
```

It will fail when run with AsciidoctorJ 2.3. This problem appears to be a regression as the test passes with 2.0.0. I've 
[attached](https://github.com/asciidoctor/asciidoctorj/files/4765302/asciidoctorj-attribute-problem.zip) a small project containing the above test. `./gradlew build` should fail.

**Comment by robertpanzer** [comment]
Thank you for the ticket!
It's a really interesting behavior, currently it looks like on the second run the extension config contains `:content_model->"attributes"` instead of `:content_model->:attributes` as in the first run.
I'll continue digging into this.

Until there is a fix it seems to work if you register the extension as a class instead of an instance:
```java
    public static class ExampleMacro extends InlineMacroProcessor {
        @Override
        public Object process(ContentNode parent, String target, Map<String, Object> attributes) {
            return createPhraseNode(parent, "quoted", attributes.toString(), attributes, new HashMap<>());
        }
    }

    @Test
    public void inlineMacroAttributes() {
        Asciidoctor asciidoctor = Asciidoctor.Factory.create();
        asciidoctor.javaExtensionRegistry().inlineMacro("example", ExampleMacro.class);
        ...
```

**Comment by wilkinsona** [comment]
Thanks for taking a look and for the workaround. Unfortunately,I need to pass something into the constructor of my real processor so I can't register it is a class. For now, I'm working around the problem by parsing things manually when `text` is present in the attributes.

**Comment by robertpanzer** [comment]
I created #927 with a fix and also copied your test into it.
Would you be able to test this?

**Comment by wilkinsona** [comment]
I tried to build your branch locally (with `./gradlew pTML`) and it failed:

```
> Task :asciidoctorj:signPom FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':asciidoctorj:signPom'.
> Cannot perform signing task ':asciidoctorj:signPom' because it has no configured signatory

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 10s
10 actionable tasks: 10 executed
```

**Comment by abelsromero** [comment]
You can pass the gradle property "skip.signing" https://github.com/asciidoctor/asciidoctorj/blob/e24117322049950a92daa46b9c04252fde78b653/gradle/signing.gradle#L1 to avoid that for a local build.

We do need to highlight this somewehere, it's not the first time someone reports struggling with this.

**Comment by wilkinsona** [comment]
Thank you, @abelsromero. `./gradlew -Pskip.signing pTML` gave me a successful build. I'll test the locally built snapshot and report back later today hopefully.

**Comment by wilkinsona** [comment]
The fix looks good to me. I removed our workaround and rebuilt our documentation. The problem reoccured as expected. I then configured the Asciidoctor Gradle plugin to use AsciidoctorJ 2.4.0-SNAPSHOT from my local Maven repository. The problem then went away again. Thanks for the quick fix.

**Comment by robertpanzer** [comment]
:+1: Awesome, thank you for testing this out.
I'll create a release 2.3.1 this week.

**Comment by robertpanzer** [comment]
I just released 2.3.1 which should contain the fix.

**Comment by wilkinsona** [comment]
Thanks very much.
